MAGECLOUD-3054: Implement Database Locks using Zookeeper and flock for Reliable HA Support (#4137)
magento/devdocs: guides/v2.2/install-gde/install/cli/install-cli-subcommands-lock.md

---
group: installation-guide
title: Configure the lock
functional_areas:
  - Install
  - System
  - Setup
---

## First steps {#instgde-cli-before}
{% include install/first-steps-cli.md %}
In addition to the command arguments discussed here, see [Common arguments]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands.html#instgde-cli-subcommands-common).

## Prerequisites {#instgde-cli-subcommands-store-prereq}

Before you run this command, you must do all of the following *or* you must [install the Magento software]({{ page.baseurl }}/install-gde/install/cli/install-cli-install.html):

*	[Create or update the deployment configuration]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-deployment.html)
*	[Create the Magento database schema]({{ page.baseurl }}/install-gde/install/cli/install-cli-subcommands-db.html)

{% include install/fully-secure.md %}

## Configure the lock {#instgde-cli-lockconfig}

> Since Magento 2.2.9 and 2.3.2 you can configure lock provider.

Reviewer: ```suggestion
Configure a lock provider to prevent the launch of duplicate cron jobs and cron groups. (Requires Magento 2.2.9 and later 2.2.x versions or version 2.3.2 and later.)
```

> Locks are used to prevent the launch of duplicate cron jobs and cron groups.

Reviewer: ```suggestion
Magento uses the database to save locks by default. If you have multiple nodes on your Magento Commerce servers, we recommend using Zookeeper as the lock provider.

If you are running Magento Commerce on the cloud infrastructure, you do not need to configure lock provider settings. Magento configures the file lock provider for Magento Commerce Pro projects during the provisioning process. See [Cloud variables]({{ page.baseurl }}/cloud/env/variables-cloud.html).
```

Command usage:

	magento setup:store-config:set [--<parameter_name>=<value>, ...]

where the following table defines parameters and values.

|Name|Value|Required?|
|--- |--- |--- |
|`--lock-provider`|Lock provider name.<br><br>Available lock providers: `db`, `zookeeper`, `file`.<br><br>The default lock provider: `db`|No|

> |`--lock-db-prefix`|The specific db prefix to avoid lock conflicts when using `db` lock provider.<br><br>The default value: `NULL`|No|

Reviewer: ```suggestion
|`--lock-db-prefix`|The specific db prefix to avoid lock conflicts when using the `db` lock provider.<br><br>The default value: `NULL`|No|
```

> |`--lock-zookeeper-host`|Host and port to connect to Zookeeper cluster when you use `zookeeper` lock provider.<br><br>For example: `127.0.0.1:2181`|Yes, if you set `--lock-provider=zookeeper`|

Reviewer: ```suggestion
|`--lock-zookeeper-host`|Host and port to connect to the Zookeeper cluster when you use the `zookeeper` lock provider.<br><br>For example: `127.0.0.1:2181`|Yes, if you set `--lock-provider=zookeeper`|
```

> |`--lock-zookeeper-path`|The path where Zookeeper will save locks.<br><br>The default path is: `/magento/locks`|No|

Reviewer: ```suggestion
|`--lock-zookeeper-path`|The path where Zookeeper saves locks.<br><br>The default path is: `/magento/locks`|No|
```

> |`--lock-file-path`|The path where file locks will be saved.|Yes, if you set `--lock-provider=file`|

Reviewer: ```suggestion
|`--lock-file-path`|The path where file locks are saved.|Yes, if you set `--lock-provider=file`|
```
